internal/temporalcli/commands.config.go

```diff
@@ -23,9 +22,18 @@ func (c *TemporalConfigDeleteCommand) run(cctx *CommandContext, _ []string) erro
 	if strings.HasPrefix(c.Prop, "grpc_meta.") {
 		key := strings.TrimPrefix(c.Prop, "grpc_meta.")
 		if _, ok := confProfile.GRPCMeta[key]; !ok {
-			return fmt.Errorf("gRPC meta key %q not found", key)
+			return fmt.Errorf("property %q not found", c.Prop)
 		}
 		delete(confProfile.GRPCMeta, key)
+	} else if strings.HasPrefix(c.Prop, "oauth.request_params.") {
+		key := strings.TrimPrefix(c.Prop, "oauth.request_params.")
+		if confProfile.OAuth == nil || confProfile.OAuth.RequestParams == nil {
+			return fmt.Errorf("property %q not found", c.Prop)
+		}
+		if _, ok := confProfile.OAuth.RequestParams[key]; !ok {
+			return fmt.Errorf("property %q not found", c.Prop)
+		}
+		delete(confProfile.OAuth.RequestParams, key)
```

Comment:
I'm going with the assumption that any profile property should be treated equally; however, it seems unlikely that anyone would meddle with these OAuth params esp. the OAuth RequestParams.